src/components/scenes/TransactionListScene.tsx

```diff
@@ -229,26 +233,30 @@ class TransactionListComponent extends React.PureComponent<Props, State> {
   }
 }
 
-export const TransactionList = connect<StateProps, DispatchProps, OwnProps>(
-  (state, ownProps) => {
-    const { walletId, currencyCode } = ownProps.route.params
-
-    // getTransactions
-    const { currencyWallets } = state.core.account
-    const currencyWallet = currencyWallets[walletId]
-    const tokenId = currencyWallet != null ? getTokenId(state.core.account, currencyWallet.currencyInfo.pluginId, currencyCode) : undefined
-
-    return {
-      numTransactions: state.ui.scenes.transactionList.numTransactions,
-      currencyCode,
-      wallet: currencyWallet,
-      tokenId,
-      transactions: state.ui.scenes.transactionList.transactions
-    }
-  },
-  dispatch => ({
-    fetchMoreTransactions(walletId: string, currencyCode: string, reset: boolean) {
-      dispatch(fetchMoreTransactions(walletId, currencyCode, reset))
-    }
+export const TransactionList = withWallet((props: OwnProps) => {
+  const { wallet, route } = props
+  const { walletId, currencyCode } = route.params
+  const theme = useTheme()
+  const dispatch = useDispatch()
+
+  const account = useSelector(state => state.core.account)
+  const numTransactions = useSelector(state => state.ui.scenes.transactionList.numTransactions)
+  const transactions = useSelector(state => state.ui.scenes.transactionList.transactions)
+  const tokenId = getTokenId(account, wallet.currencyInfo.pluginId, currencyCode)
+
+  const handleMoreTransactions = useHandler((reset: boolean): void => {
+    dispatch(fetchMoreTransactions(walletId, currencyCode, reset))
```

Comment:
I suspect using these route params is fixed in a later commit but leaving a comment so I don't forget to check